example_code/shmem_team_context.c

```diff
@@ -1,106 +1,90 @@
 #include <shmem.h>
 #include <stdio.h>
 
-int isum, ival;
 
-int my_ctx_translate_pe(shmem_ctx_t src_ctx, int src_pe, shmem_ctx_t dest_ctx) {
-  if (src_ctx == SHMEM_CTX_INVALID) {
-    return -1;
-  }
-  if (dest_ctx == SHMEM_CTX_INVALID) {
-    return -1;
-  }
-
-  shmem_team_t src_team, dest_team;
-  shmem_ctx_get_team(src_ctx, &src_team);
-  shmem_ctx_get_team(dest_ctx, &dest_team);
-  return shmem_team_translate_pe(src_team, src_pe, dest_pe);
-}
-
-shmem_ctx_t my_team_create_ctx(shmem_team_t team) {
-  if (team == SHMEM_TEAM_INVALID) {
-    return SHMEM_CTX_INVALID;
-  }
+int main(void)
+{
+  static int            sum = 0, val_2, val_3;
+  shmem_team_t          team_2, team_3;
+  shmem_ctx_t           ctx_2, ctx_3;
+  shmem_team_config_t   conf;
 
-  shmem_ctx_t ctx;
-  if (shmem_team_create_ctx(team, 0, &ctx) != 0) {
-    fprintf(stderr, "Failed to create context for PE team\n");
-    return SHMEM_CTX_INVALID;
-  }
-  return ctx;
-}
+  shmem_init();
 
-void my_send_to_neighbor(shmem_ctx_t ctx, int *val) {
-  if (ctx == SHMEM_CTX_INVALID) {
-    fprintf(stderr, "Send to neighbor fail due to invalid context\n");
-    return;
-  }
+  int npes = shmem_n_pes();
+  int mype = shmem_my_pe();
+  conf.num_contexts = 1;
+  long cmask = SHMEM_TEAM_NUM_CONTEXTS;
 
-  shmem_team_t team;
-  shmem_ctx_get_team(ctx, &team);
-  int team_mype = shmem_team_my_pe(team);
-  int team_npes = shmem_team_n_pes(team);
-  int rpe = (team_mype + 1) % team_npes;
+  /* Create team_2 with PEs numbered 0, 2, 4, ... */
+  shmem_team_split_strided(SHMEM_TEAM_WORLD, 0, 2, (npes + 1) / 2, &conf, cmask, &team_2);
```

Comment:
Do we need to check the return code from the split operations? Or is having the team set to `SHMEM_TEAM_INVALID` sufficient?

Reply:
I certainly don't think it hurts to check the return value here - it might encourage good practice.  Is that what you prefer?

Reply:
It's probably good practice for an example.